shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/TableMetaDataBuilder.java

```diff
@@ -55,73 +55,52 @@
-        return tableMetaData.map(optional -> decorate(tableName, optional, materials.getRules()));
+        return Optional.ofNullable(load(Collections.singleton(tableName), materials).getOrDefault(tableName, null)).map(tableMetaData -> decorate(tableName, tableMetaData, materials.getRules()));
     }
     
     /**
-     * Load physical table metadata.
-     * 
-     * @param tableName table name
-     * @param materials schema builder materials
-     * @return table meta data
-     * @throws SQLException SQL exception
-     */
-    @SuppressWarnings({"unchecked", "rawtypes"})
-    public static Optional<TableMetaData> load(final String tableName, final SchemaBuilderMaterials materials) throws SQLException {
-        DataNodes dataNodes = new DataNodes(materials.getRules());
-        for (Entry<ShardingSphereRule, RuleBasedTableMetaDataBuilder> entry : OrderedSPIRegistry.getRegisteredServices(RuleBasedTableMetaDataBuilder.class, materials.getRules()).entrySet()) {
-            if (entry.getKey() instanceof TableContainedRule) {
-                TableContainedRule rule = (TableContainedRule) entry.getKey();
-                RuleBasedTableMetaDataBuilder loader = entry.getValue();
-                Optional<TableMetaData> result = loader.load(tableName, materials.getDatabaseType(), materials.getDataSourceMap(), dataNodes, rule, materials.getProps());
-                if (result.isPresent()) {
-                    TableMetaData tableMetaData = new TableMetaData(tableName, result.get().getColumns().values(), result.get().getIndexes().values());
-                    return Optional.of(tableMetaData);
-                }
-            }
-        }
-        return Optional.empty();
-    }
-    
-    /**
-     * Load logic table metadata.
+     * Load all table metadata.
      *
+     * @param tableNames table name collection
      * @param materials schema builder materials
-     * @param executorService executorService
-     * @return table meta data collection
+     * @return table meta data map
      * @throws SQLException SQL exception
      */
     @SuppressWarnings("rawtypes")
-    public static Collection<TableMetaData> loadLogicTables(final SchemaBuilderMaterials materials, final ExecutorService executorService) throws SQLException {
-        Collection<TableMetaData> result = new LinkedList<>();
+    public static Map<String, TableMetaData> load(final Collection<String> tableNames, final SchemaBuilderMaterials materials) throws SQLException {
+        Map<String, TableMetaData> result = new LinkedHashMap<>();
         for (Entry<ShardingSphereRule, RuleBasedTableMetaDataBuilder> entry : OrderedSPIRegistry.getRegisteredServices(RuleBasedTableMetaDataBuilder.class, materials.getRules()).entrySet()) {
             if (entry.getKey() instanceof TableContainedRule) {
-                loadTableContainedRuleTables(materials, executorService, result, entry);
+                loadTableContainedRuleTables(tableNames, materials, result, entry);
```

Comment:
@tuichenchuxin The method is called loadTableContainedRuleTables, and it is better to have a return value instead of returning void.